api/openapi.yaml

```diff
@@ -446,30 +481,56 @@ components:
           description: Example "POST /users/verify"
         valueType:
           type: string
-          description: TODO(pov) think about value types?
-          enum: [stringValue, intValue]
+          enum:
+            [
+              stringValue,
+              intValue,
+              boolValue,
+              doubleValue,
+              kvlistValue,
+            ]
     SpanAssertion:
       type: object
+      required:
+        - spanAssertionId
+        - locationName
+        - propertyName
+        - valueType
+        - operator
+        - comparisonValue
       properties:
         spanAssertionId:
           type: string
           description: ID
           readOnly: true
         locationName:
           type: string
-          enum: [RESOURCE_ATTRIBUTES, INSTRUMENTATION_LIBRARY, SPAN, SPAN_ATTRIBUTES]
+          enum:
+            [
+              RESOURCE_ATTRIBUTES,
+              INSTRUMENTATION_LIBRARY,
+              SPAN,
+              SPAN_ATTRIBUTES,
+            ]
           description: Example "SPAN_ATTRIBUTES"
         propertyName:
           type: string
           description: Example "http.status.code"
         valueType:
           type: string
           description: Example "intValue"
-          enum: [stringValue, intValue]
+          enum:
+            [
+              stringValue,
+              intValue,
+              boolValue,
+              doubleValue,
+              kvlistValue,
+            ]
         operator:
           type: string
           description: Example "EQUALS"
-          enum: [EQUALS, LESSTHAN, GREATERTHAN, NOTEQUALS]
+          enum: [EQUALS, NOTEQUALS, LESSTHAN, GREATERTHAN, GREATOREQUALS, LESSOREQUAL, CONTAINS]
```

Comment:
NOTE: This is fine, but it will become obsolete once we move the assertion engine to the backend.

Reply:
Yep, I did this to match what we currently have in the FE